.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,40 +1,63 @@
 ## PR Summary
 
-## PR Checklist
+<!--
+Thank you so much for your PR! To help us review your contribution, please
+summarize the purpose of this PR here.
 
-- [ ] Has Pytest style unit tests
-- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
-- [ ] New features are documented, with examples if plot related
-- [ ] Documentation is sphinx and numpydoc compliant
-- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
-- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way
+- The summary should provide at least 1-2 sentences describing the pull request
+  in detail (Why is this change required? What problem does it solve?) and
+  link to any relevant issues.
+-->
 
-<!--
-Thank you so much for your PR!  To help us review your contribution, please
-consider the following points:
 
-- A development guide is available at https://matplotlib.org/devdocs/devel/index.html.
+## PR Checklists
 
-- Help with git and github is available at
-  https://matplotlib.org/devel/gitwash/development_workflow.html.
+<!-- Feel free to delete any checklists that do not apply to this PR -->
+
+For new code:
+
+- [ ] Has Pytest style unit tests (and `pytest lib/matplotlib/tests` passes)
+- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check)
+- [ ] New code is documented, with examples if plot related
+
+For new documentation:
+
+- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error)
+- [ ] New documentation conforms to matplotlib style conventions. (If you have `flake8-docstrings` and `pydocstyle<4` installed, run `flake8 --docstring-convention=all` on changed files to check).
+
+For new features:
+
+- [ ] Added an entry to `doc/users/next_whats_new/` if major new feature (follow instructions in `README.rst` there)
+
+For API changes:
+
+- [ ] Documented in `doc/api/api_changes_[VERSION]` if API changed in a backward-incompatible way (follow instructions in `README.rst` there)
+
+<!--
+A few other common gotchas:
 
 - Do not create the PR out of master, but out of a separate branch.
 
 - The PR title should summarize the changes, for example "Raise ValueError on
-  non-numeric input to set_xlim".  Avoid non-descriptive titles such as
+  non-numeric input to set_xlim". Avoid non-descriptive titles such as
```

Comment:
Kind think maybe take out the descriptive pieces up top and replace w/ another checklist (folks seem to pay more attention to lists than to prose)
Pull Request
[] PR has descriptive name 
[]PR has minimum of 1-2 sentence descriptive summary
[] optional:PR cross-links related issues

Reply:
So I included a "Meta" section. To be honest, I think it's a little clearer just as prose, especially since it makes the already imposing list of checklists even longer....but if we have better luck with the checkboxes I guess you gotta do what you know works....

Reply:
What about making the "Meta" section bullet points? This is more like a reminder. I don't want to tick numerous boxes when making a PR. These things are easy to apply and there is no reason not to immediately do it (in contrast to e.g. writing good tests, where sometimes it's reasonable to open a PR without.)

Same would maybe apply to "New Documentation" section?

Reply:
I agree that I don't want to be checking off boxes for every PR when it's just things like "did you fill in the summary". 

If I do make them just bullet points, then I see no need for it to ever be rendered (reviewer knows about these, don't need the visual noise), so I've changed it to be commented out by default.

Pasted latest version below.